llvm/docs/LangRef.rst

```diff
@@ -26965,13 +26966,23 @@ to facilitate alias analysis and underlying-object detection.
 Semantics:
 """"""""""
 
-The result of ``ptrmask(ptr, mask)`` is equivalent to
-``getelementptr ptr, (ptrtoint(ptr) & mask) - ptrtoint(ptr)``. Both the returned
-pointer(s) and the first argument are based on the same underlying object (for more
-information on the *based on* terminology see
-:ref:`the pointer aliasing rules <pointeraliasing>`). If the bitwidth of the
-mask argument does not match the pointer size of the target, the mask is
-zero-extended or truncated accordingly.
+The result of ``ptrmask(%ptr, %mask)`` is equivalent to the following expansion,
+where ``iPtrIdx`` is the index type size of the pointer::
```

Comment:
```suggestion
where ``iPtrIdx`` is the index type size of the pointer:
```
I think this extra ':' is not needed?

Reply:
The double colon is intentional -- it's a weird bit of rst syntax: https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#literal-blocks